docs/userGuide/usingPlugins.md

```diff
@@ -106,6 +108,37 @@ module.exports = {
 </div>
 ```
 
+#### Assets
+
+Plugins can implement the methods `getLinks` and `getScripts` to add additional assets to the page. 
+
+- `getLinks(content, pluginContext, frontMatter, utils)`: Called to get link elements to be added to the head of the page.
+  - `content`: The rendered HTML.
+  - `pluginContext`: User provided parameters for the plugin. This can be specified in the `site.json`.
+  - `frontMatter`: The frontMatter of the page being processed, in case any frontMatter data is required.
+  - `utils`: Object containing the following utility functions
+    - `buildStylesheet(href)`: Builds a stylesheet link element with the specified `href`.
+  - Should return an array of string data containing link elements to be added.
+- `getScripts(context, pluginContext, frontMatter, utils)`: Called to get script elements to be added after the body of the page.
+  - `content`: The rendered HTML.
+  - `pluginContext`: User provided parameters for the plugin. This can be specified in the `site.json`.
+  - `frontMatter`: The frontMatter of the page being processed, in case any frontMatter data is required.
+  - `utils`: Object containing the following utility functions
+    - `buildScript(src)`: Builds a script element with the specified `src`.
+  - Should return an array of string data containing script elements to be added.
+
+An example of a plugin which adds links and scripts to the page:
+
+```js
+// myPlugin.js
+
+module.exports = {
+  getLinks: (content, pluginContext, frontMatter, utils) => [utils.buildStylesheet('STYLESHEET_LINK')],
+  getScripts: (content, pluginContext, frontMatter, utils) => [utils.buildScript('SCRIPT_LINK')],

```

Comment:
Could we have an example of an inline script specified by user?

Reply:
On a side note, are there any security concerns for this. Eg: users trying malicious plugins written by attackers?

Reply:
Currently, we have a warning [written in the documentation](https://markbind.org/userGuide/usingPlugins.html) to warn authors not to run any plugins that they do not know about:

```
WARNING: Plugins are executable programs that can be written by anyone. This means that they might contain malicious code that may damage your computer.

Only run plugins from sources that you trust. Do not run the plugin if the source/origin of the plugin cannot be ascertained.
```

The content that they can modify for the MarkBind website is limited to what MarkBind passes in as parameters. However, since they are just JavaScript files, they still have access to Nodejs methods and functions.

So perhaps to strengthen the security further, we should also explore running plugins inside a sandbox. Will post an issue soon.